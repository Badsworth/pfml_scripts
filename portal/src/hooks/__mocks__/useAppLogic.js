import Claim from "../../models/Claim";
import ClaimCollection from "../../models/ClaimCollection";
import { uniqueId } from "lodash";

export default jest.fn(() => ({
  auth: {
    createAccount: jest.fn(),
    forgotPassword: jest.fn(),
    login: jest.fn(),
  },
  claims: new ClaimCollection(),
  createClaim: jest.fn(() => new Claim({ application_id: uniqueId() })),
  loadClaims: jest.fn(),
  submitClaim: jest.fn(),
  updateClaim: jest.fn(),
}));
