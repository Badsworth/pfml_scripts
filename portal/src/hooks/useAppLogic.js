/* eslint sort-keys: ["error", "asc"] */
import useAppErrorsLogic from "./useAppErrorsLogic";
import useAuthLogic from "./useAuthLogic";
import useBenefitsApplicationsLogic from "./useBenefitsApplicationsLogic";
import useClaimsLogic from "./useClaimsLogic";
import useDocumentsLogic from "./useDocumentsLogic";
import useEmployersLogic from "./useEmployersLogic";
import useFlagsLogic from "./useFeatureFlagsLogic";
import useOtherLeavesLogic from "./useOtherLeavesLogic";
import usePortalFlow from "./usePortalFlow";
import useUsersLogic from "./useUsersLogic";

const useAppLogic = () => {
  // Utility to determine application flow
  const portalFlow = usePortalFlow();

  // State representing currently visible errors and warnings
  const { appErrors, ...appErrorsLogic } = useAppErrorsLogic({ portalFlow });
  const auth = useAuthLogic({ appErrorsLogic, portalFlow });

  // State representing the Portal's user object.
  // Initialize to empty user but will be populated upon the first API call
  // to fetch the user (or create the user on their first login)
  const users = useUsersLogic({
    appErrorsLogic,
    isLoggedIn: auth.isLoggedIn,
    portalFlow,
  });

  // user will be eventually set through a `login` method
  // const [user, setUser] = useState();
  // TODO (CP-885): remove user from configs and write login method
  const benefitsApplications = useBenefitsApplicationsLogic({
    appErrorsLogic,
    portalFlow,
    user: users.user,
  });

  const claims = useClaimsLogic({
    appErrorsLogic,
  });

  const documents = useDocumentsLogic({
    appErrorsLogic,
    portalFlow,
  });

  const employers = useEmployersLogic({
    appErrorsLogic,
    portalFlow,
    setUser: users.setUser,
  });

  const otherLeaves = useOtherLeavesLogic({ appErrorsLogic });
  const featureFlags = useFlagsLogic({ appErrorsLogic });

  return {
    appErrors,
    auth,
    benefitsApplications,
    catchError: appErrorsLogic.catchError,
    claims,
    // TODO (CP-886): remove once all API calls are behind appLogic
    clearErrors: appErrorsLogic.clearErrors,
    documents,
    employers,
    otherLeaves,
    portalFlow,
    setAppErrors: appErrorsLogic.setAppErrors,
    users,
    featureFlags,
  };
};

export default useAppLogic;
