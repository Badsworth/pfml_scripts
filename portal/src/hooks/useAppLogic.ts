/* eslint sort-keys: ["error", "asc"] */
import useApplicationImportsLogic from "./useApplicationImportsLogic";
import useAuthLogic from "./useAuthLogic";
import useBenefitYearsLogic from "./useBenefitYearsLogic";
import useBenefitsApplicationsLogic from "./useBenefitsApplicationsLogic";
import useClaimsLogic from "./useClaimsLogic";
import useDocumentsLogic from "./useDocumentsLogic";
import useEmployersLogic from "./useEmployersLogic";
import useErrorsLogic from "./useErrorsLogic";
import useFeatureFlagsLogic from "./useFeatureFlagsLogic";
import useHolidaysLogic from "./useHolidaysLogic";
import usePaymentsLogic from "./usePaymentsLogic";
import usePortalFlow from "./usePortalFlow";
import useUsersLogic from "./useUsersLogic";

const useAppLogic = () => {
  // Utility to determine application flow
  const portalFlow = usePortalFlow();

  // State representing currently visible errors and warnings
  const errorsLogic = useErrorsLogic({ portalFlow });

  const applicationImports = useApplicationImportsLogic({
    errorsLogic,
    portalFlow,
  });
  const auth = useAuthLogic({ errorsLogic, portalFlow });

  // State representing the Portal's user object.
  // Initialize to empty user but will be populated upon the first API call
  // to fetch the user (or create the user on their first login)
  const users = useUsersLogic({
    errorsLogic,
    isLoggedIn: !!auth.isLoggedIn,
    portalFlow,
  });

  const benefitYears = useBenefitYearsLogic({ errorsLogic });

  const benefitsApplications = useBenefitsApplicationsLogic({
    errorsLogic,
    portalFlow,
  });

  const claims = useClaimsLogic({
    errorsLogic,
    portalFlow,
  });

  const documents = useDocumentsLogic({
    errorsLogic,
  });

  const employers = useEmployersLogic({
    clearClaims: claims.clearClaims,
    errorsLogic,
    portalFlow,
    setUser: users.setUser,
  });

  const featureFlags = useFeatureFlagsLogic();

  const holidays = useHolidaysLogic({
    errorsLogic,
  });

  const payments = usePaymentsLogic({
    errorsLogic,
  });

  return {
    // `_errorsLogic` should not be used except for testing
    _errorsLogic: errorsLogic,
    applicationImports,
    auth,
    benefitYears,
    benefitsApplications,
    catchError: errorsLogic.catchError,
    claims,
    // TODO (CP-886): remove once all API calls are behind appLogic
    clearErrors: errorsLogic.clearErrors,
    clearRequiredFieldErrors: errorsLogic.clearRequiredFieldErrors,
    documents,
    employers,
    errors: errorsLogic.errors,
    featureFlags,
    holidays,
    payments,
    portalFlow,
    setErrors: errorsLogic.setErrors,
    users,
  };
};

export default useAppLogic;
export type AppLogic = ReturnType<typeof useAppLogic>;
