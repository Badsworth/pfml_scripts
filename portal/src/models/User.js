import BaseModel from "./BaseModel";

class User extends BaseModel {
  get defaults() {
    return {
      user_id: null,
      cognito_user_id: null,
      first_name: null,
      middle_name: null,
      last_name: null,
      date_of_birth: null,
      has_state_id: null,
      state_id: null,
      status: null,
      ssn_or_itin: null,
    };
  }
}

export default User;
