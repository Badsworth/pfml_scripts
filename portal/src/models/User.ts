/* eslint sort-keys: ["error", "asc"] */
export const PhoneType = {
  cell: "Cell",
  phone: "Phone",
} as const;

export const MFAPreference = {
  optOut: "Opt Out",
  sms: "SMS",
} as const;

class User {
  auth_id: string;
  consented_to_data_sharing: boolean;
  email_address: string;
  roles: UserRole[] = [];
  user_id: string;
  mfa_phone_number: {
    int_code: string | null;
    phone_number: string | null;
    phone_type: typeof PhoneType[keyof typeof PhoneType] | null;
  } | null = null;

  mfa_delivery_preference:
    | typeof MFAPreference[keyof typeof MFAPreference]
    | null = null;

  user_leave_administrators: UserLeaveAdministrator[] = [];

  constructor(attrs: Partial<User>) {
    Object.assign(this, attrs);
  }

  get hasEmployerRole() {
    return this.roles.some(
      (userRole) => userRole.role_description === RoleDescription.employer
    );
  }

  /**
   * Determines whether user_leave_administrators has only unverified employers
   */
  get hasOnlyUnverifiedEmployers(): boolean {
    return this.verifiedEmployers.length === 0;
  }

  /**
   * Determines whether user_leave_administrators has a verifiable employer
   */
  get hasVerifiableEmployer(): boolean {
    return this.user_leave_administrators.some((employer) =>
      this.isVerifiableEmployer(employer)
    );
  }

  /**
   * Returns list of verified employers
   */
  get verifiedEmployers(): UserLeaveAdministrator[] {
    return this.user_leave_administrators.filter(
      (employer) => employer.verified === true
    );
  }

  /**
   * Determines whether an employer is verifiable (unverified and has verification data)
   */
  isVerifiableEmployer(employer: UserLeaveAdministrator): boolean {
    return !employer.verified && employer.has_verification_data;
  }

  /**
   * Determines whether an employer is registered in FINEOS
   */
  isEmployerIdRegisteredInFineos(employerId: string | null): boolean {
    return this.user_leave_administrators.some(
      (employer) =>
        employerId === employer.employer_id && employer.has_fineos_registration
    );
  }
}

export class UserRole {
  role_description: typeof RoleDescription[keyof typeof RoleDescription];
  role_id: number;

  constructor(attrs: Partial<UserRole>) {
    Object.assign(this, attrs);
  }
}

/**
 * Enums for UserRole's `role_description` field
 * @enum {string}
 */
export const RoleDescription = {
  claimant: "Claimant",
  employer: "Employer",
} as const;

export class UserLeaveAdministrator {
  employer_dba: string;
  employer_fein: string;
  employer_id: string;
  has_fineos_registration: boolean;
  has_verification_data: boolean;
  verified: boolean;

  constructor(attrs: Partial<UserLeaveAdministrator>) {
    Object.assign(this, attrs);
  }
}

export default User;
