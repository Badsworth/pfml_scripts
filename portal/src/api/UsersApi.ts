/* eslint-disable jsdoc/require-returns */
import User, { UserLeaveAdministrator, UserRole } from "../models/User";

import BaseApi from "./BaseApi";
import routes from "../routes";

/**
 * @typedef {{ user: User }} UsersApiResult
 * @property {User} [user] - If the request succeeded, this will contain the created user
 */

export default class UsersApi extends BaseApi {
  // @ts-expect-error ts-migrate(2416) FIXME: Property 'basePath' in type 'UsersApi' is not assi... Remove this comment to see the full error message
  get basePath() {
    return routes.api.users;
  }

  // @ts-expect-error ts-migrate(2416) FIXME: Property 'i18nPrefix' in type 'UsersApi' is not as... Remove this comment to see the full error message
  get i18nPrefix() {
    return "users";
  }

  /**
   * Register a new user
   * @param {object} requestData - Registration fields
   * @returns {Promise<UsersApiResult>}
   */
  createUser = async (requestData) => {
    const { data } = await this.request(
      "POST",
      "",
      requestData,
      {},
      { excludeAuthHeader: true }
    );

    return Promise.resolve({
      user: new User(data),
    });
  };

  /**
   * Get the currently authenticated user
   * @returns {Promise<UsersApiResult>}
   */
  getCurrentUser = async () => {
    const { data } = await this.request("GET", "current", null);
    const roles = this.createUserRoles(data.roles);
    const user_leave_administrators = this.createUserLeaveAdministrators(
      data.user_leave_administrators
    );

    return Promise.resolve({
      user: new User({ ...data, roles, user_leave_administrators }),
    });
  };

  /**
   * Update a user
   * @param {object} user_id - ID of user being updated
   * @param {object} patchData - User fields to update
   * @returns {Promise<UsersApiResult>}
   */
  updateUser = async (user_id, patchData) => {
    const { data } = await this.request("PATCH", user_id, patchData);
    const roles = this.createUserRoles(data.roles);
    const user_leave_administrators = this.createUserLeaveAdministrators(
      data.user_leave_administrators
    );

    return {
      user: new User({
        ...patchData,
        ...data,
        roles,
        user_leave_administrators,
      }),
    };
  };

  /**
   * Convert a user role to another
   * @param {string} user_id - ID of user being converted
   * @param {object} postData - User fields to update - role and leave admin
   * @returns {Promise<UsersApiResult>}
   */
  convertUser = async (user_id, postData) => {
    const { data } = await this.request(
      "POST",
      `${user_id}/convert_employer`,
      postData
    );
    const roles = this.createUserRoles(data.roles);
    const user_leave_administrators = this.createUserLeaveAdministrators(
      data.user_leave_administrators
    );

    return {
      user: new User({
        ...data,
        roles,
        user_leave_administrators,
      }),
    };
  };

  createUserLeaveAdministrators = (leaveAdmins) => {
    return (leaveAdmins || []).map(
      (leaveAdmin) => new UserLeaveAdministrator(leaveAdmin)
    );
  };

  createUserRoles = (roles) => {
    return (roles || []).map((role) => new UserRole(role));
  };
}
