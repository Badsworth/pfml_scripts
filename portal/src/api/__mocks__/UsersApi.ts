/**
 * @file Mock Users API module for tests. Jest requires the file
 * to be adjacent to the module we're mocking.
 * @see https://jestjs.io/docs/en/manual-mocks#mocking-user-modules
 * @example jest.mock("../api/UsersApi")
 */
import User, {
  RoleDescription,
  UserLeaveAdministrator,
  UserRole,
} from "../../models/User";

const createMockUser = () =>
  new User({
    auth_id: "cognito-123",
    consented_to_data_sharing: true,
    email_address: "mock-user@example.com",
    user_id: "api-123",
  });

const createMockEmployer = () => {
  const user = createMockUser();
  // @ts-expect-error ts-migrate(2339) FIXME: Property 'roles' does not exist on type 'User'.
  user.roles.push(new UserRole({ role_description: RoleDescription.employer }));
  // @ts-expect-error ts-migrate(2339) FIXME: Property 'user_leave_administrators' does not exis... Remove this comment to see the full error message
  user.user_leave_administrators.push(
    new UserLeaveAdministrator({
      employer_dba: "Test Company",
      employer_fein: "1298391823",
      employer_id: "dda903f-f093f-ff900",
      has_verification_data: true,
      verified: true,
    })
  );
  return user;
};

// @ts-expect-error ts-migrate(2304) FIXME: Cannot find name 'jest'.
export const mockCreateUser = jest.fn(() =>
  Promise.resolve({
    success: true,
    user: createMockUser(),
  })
);

// @ts-expect-error ts-migrate(2304) FIXME: Cannot find name 'jest'.
export const mockGetCurrentUser = jest.fn(() =>
  Promise.resolve({
    success: true,
    user: createMockUser(),
  })
);

// @ts-expect-error ts-migrate(2304) FIXME: Cannot find name 'jest'.
export const mockUpdateUser = jest.fn(() =>
  Promise.resolve({
    success: true,
    user: createMockUser(),
  })
);

// @ts-expect-error ts-migrate(2304) FIXME: Cannot find name 'jest'.
export const mockConvertUser = jest.fn(() =>
  Promise.resolve({
    success: true,
    user: createMockEmployer(),
  })
);

// @ts-expect-error ts-migrate(2304) FIXME: Cannot find name 'jest'.
export default jest.fn().mockImplementation(() => ({
  createUser: mockCreateUser,
  getCurrentUser: mockGetCurrentUser,
  updateUser: mockUpdateUser,
  convertUser: mockConvertUser,
}));
