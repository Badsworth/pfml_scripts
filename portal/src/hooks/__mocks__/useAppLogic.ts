import {
  BenefitsApplicationDocument,
  ClaimDocument,
} from "../../models/Document";
import User, { UserLeaveAdministrator } from "../../models/User";
import ApiResourceCollection from "../../models/ApiResourceCollection";
import BenefitsApplication from "../../models/BenefitsApplication";
import Claim from "../../models/Claim";
import ClaimDetail from "../../models/ClaimDetail";
import EmployerClaim from "../../models/EmployerClaim";
import Flag from "../../models/Flag";
import { Payment } from "src/models/Payment";
import { uniqueId } from "lodash";

export default jest.fn(() => ({
  errors: [],
  auth: {
    createAccount: jest.fn(),
    createEmployerAccount: jest.fn(),
    forgotPassword: jest.fn(),
    isLoggedIn: true,
    login: jest.fn(),
    logout: jest.fn(),
    requireLogin: jest.fn(),
    resendForgotPasswordCode: jest.fn(),
    resendVerifyAccountCode: jest.fn(),
    resetPassword: jest.fn(),
    verifyAccount: jest.fn(),
  },
  catchError: jest.fn(),
  benefitsApplications: {
    benefitsApplications: new ApiResourceCollection<BenefitsApplication>(
      "application_id"
    ),
    complete: jest.fn(),
    create: jest.fn(
      () => new BenefitsApplication({ application_id: uniqueId() })
    ),
    get: jest.fn(),
    hasLoadedBenefitsApplicationAndWarnings: jest.fn().mockReturnValue(true),
    hasLoadedAll: false,
    load: jest.fn(),
    loadAll: jest.fn(),
    submit: jest.fn(),
    submitPaymentPreference: jest.fn(),
    update: jest.fn(),
    warningsLists: {},
  },
  claims: {
    claimDetail: new ClaimDetail(),
    claims: new ApiResourceCollection<Claim>("fineos_absence_id"),
    clearClaims: jest.fn(),
    isLoadingClaims: null,
    isLoadingClaimDetail: null,
    loadClaimDetail: jest.fn(),
    paginationMeta: {},
    loadPage: jest.fn(),
  },
  clearErrors: jest.fn(),
  clearRequiredFieldErrors: jest.fn(),
  documents: {
    attach: jest.fn((application_id, files) => {
      const uploadPromises: Array<Promise<{ success: boolean }>> = [];
      for (let i = 0; i < files.length; i++) {
        uploadPromises.push(Promise.resolve({ success: true }));
      }
      return uploadPromises;
    }),
    hasLoadedClaimDocuments: jest.fn(),
    documents: new ApiResourceCollection<BenefitsApplicationDocument>(
      "fineos_document_id"
    ),
    download: jest.fn(),
    loadAll: jest.fn(),
  },
  employers: {
    addEmployer: jest.fn(),
    downloadDocument: jest.fn(() => new Blob()),
    loadClaim: jest.fn(
      () =>
        new EmployerClaim({
          absence_periods: [],
          fineos_absence_id: "NTN-111-ABS-01",
        })
    ),
    loadDocuments: jest.fn(
      () => new ApiResourceCollection<ClaimDocument>("fineos_document_id")
    ),
    loadWithholding: jest.fn(() => ({ filing_period: "2011-11-20" })),
    submitClaimReview: jest.fn(),
    submitWithholding: jest.fn(),
  },
  payments: {
    loadPayments: jest.fn(),
    loadedPaymentsData: new Payment(),
    hasLoadedPayments: true,
    isLoadingPayments: false,
  },
  portalFlow: {
    getNextPageRoute: jest.fn(),
    goTo: jest.fn(),
    goToNextPage: jest.fn(),
    goToPageFor: jest.fn(),
    pathname: "/mock-portalFlow-pathname",
  },
  otherLeaves: {
    removeEmployerBenefit: jest.fn(() => true),
    removeOtherIncome: jest.fn(() => true),
    removePreviousLeave: jest.fn(() => true),
  },
  setErrors: jest.fn(),
  updateUser: jest.fn(),
  user: new User({
    user_id: "mock_user_id",
    consented_to_data_sharing: true,
    email_address: "ali@miau.com",
    auth_id: "cognito_123",
    roles: [],
    user_leave_administrators: [],
  }),
  users: {
    loadUser: jest.fn(),
    requireUserConsentToDataAgreement: jest.fn(),
    requireUserRole: jest.fn(),
    updateUser: jest.fn(),
    user: new User({
      auth_id: "mock_auth_id",
      user_id: "mock_user_id",
      consented_to_data_sharing: true,
      user_leave_administrators: [
        // not verified, but can be
        new UserLeaveAdministrator({
          employer_dba: "Book Bindings 'R Us",
          employer_fein: "**-***1823",
          employer_id: "dda903f-f093f-ff900",
          has_fineos_registration: true,
          has_verification_data: true,
          verified: false,
        }),
        // already verified
        new UserLeaveAdministrator({
          employer_dba: "Knitting Castle",
          employer_fein: "**-***3443",
          employer_id: "dda930f-93jfk-iej08",
          has_fineos_registration: true,
          has_verification_data: true,
          verified: true,
        }),
        // not verified and cannot be verified
        new UserLeaveAdministrator({
          employer_dba: "Tomato Touchdown",
          employer_fein: "**-***7192",
          employer_id: "io19fj9-00jjf-uiw3r",
          has_fineos_registration: true,
          has_verification_data: false,
          verified: false,
        }),
      ],
    }),
  },
  featureFlags: {
    flags: [
      new Flag({
        enabled: false,
        name: "maintenance",
        options: {
          page_routes: ["/*"],
        },
        start: null,
        end: null,
      }),
    ],
    getFlag: jest.fn(() => {
      return new Flag({
        enabled: false,
        name: "maintenance",
        options: {
          page_routes: ["/*"],
        },
        start: null,
        end: null,
      });
    }),
    loadFlags: jest.fn(),
  },
}));
