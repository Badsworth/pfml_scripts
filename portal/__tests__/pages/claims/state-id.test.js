import React from "react";
import StateId from "../../../src/pages/claims/state-id";
import User from "../../../src/models/User";
import { act } from "react-dom/test-utils";
import { mockUpdateUser } from "../../../src/api/UsersApi";
import { shallow } from "enzyme";
import { testHook } from "../../test-utils";
import useAppLogic from "../../../src/hooks/useAppLogic";

jest.mock("../../../src/api/UsersApi");

const claim_id = "12345";
const render = (props = {}) => {
  let appLogic;

  testHook(() => {
    appLogic = useAppLogic();
  });

  appLogic.user = props.user || new User();

  const allProps = {
    appLogic,
    query: { claim_id },
    claim: {},
    ...props,
  };
  return {
    props: allProps,
    wrapper: shallow(<StateId {...allProps} />),
  };
};

describe("StateId", () => {
  let wrapper;

  beforeEach(() => {
    ({ wrapper } = render());
  });

  it("initially renders the page without an id field", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("ConditionalContent").prop("visible")).toBeFalsy();
  });

  describe("when user has a state id", () => {
    it("renders id field", () => {
      const user = new User({
        has_state_id: true,
        state_id: "12345",
      });
      ({ wrapper } = render({ user }));

      expect(
        wrapper.update().find("ConditionalContent").prop("visible")
      ).toBeTruthy();
    });
  });

  describe("when the form is successfully submitted", () => {
    it("calls updateUser and updates the state", async () => {
      expect.assertions();
      const { wrapper: component } = render();
      await act(async () => {
        await component.find("QuestionPage").simulate("save");
      });

      expect(mockUpdateUser).toHaveBeenCalledTimes(1);
    });
  });
});
