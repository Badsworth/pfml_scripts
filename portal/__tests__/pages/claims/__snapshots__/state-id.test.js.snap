// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateId initially renders the page without an id field 1`] = `
<QuestionPage
  formState={
    User {
      "cognito_user_id": null,
      "date_of_birth": null,
      "first_name": null,
      "has_state_id": null,
      "last_name": null,
      "middle_name": null,
      "ssn_or_itin": null,
      "state_id": null,
      "status": null,
      "user_id": null,
    }
  }
  nextPage="/claims/ssn"
  onSave={[Function]}
  title="Who is taking leave"
>
  <InputChoiceGroup
    choices={
      Array [
        Object {
          "checked": false,
          "label": "Yes",
          "value": "true",
        },
        Object {
          "checked": false,
          "label": "No",
          "value": "false",
        },
      ]
    }
    label="Do you have a Massachusetts driver's license or ID card?"
    name="has_state_id"
    onChange={[Function]}
    type="radio"
  />
  <ConditionalContent
    fieldNamesClearedWhenHidden={
      Array [
        "state_id",
      ]
    }
    removeField={[Function]}
    visible={false}
  >
    <InputText
      label="Enter your license or ID number"
      name="state_id"
      onChange={[Function]}
      value=""
    />
  </ConditionalContent>
</QuestionPage>
`;
